source/fe/fe_wedge_p.cc

```diff
@@ -103,14 +103,75 @@ FE_WedgePoly<dim, spacedim>::FE_WedgePoly(
 {
   AssertDimension(dim, 3);
 
-  if (degree == 1)
+  Assert(1 <= degree && degree <= 2, ExcNotImplemented())
+
+    for (double z = 0.0; z <= 1.0; z += (degree == 1 ? 1.0 : 0.5))
```

Comment:
```suggestion
    for (double z = 0.0; z <= 1.0; z += 1./degree)
```
This is going to run into trouble for values of `degree` for which `1./degree` is not exactly representable in floating point arithmetic. The test `z<=1` will then either work or not work, by accident. It's better to write this as
```
  for (unsigned int layer=0; layer<=degree; ++layer)
  {
     const double delta_z = 1./degree;
     const double z = layer * delta_z;
     ...
```